app/src/main/java/com/example/tasky/feature_agenda/data/worker/WorkerUtil.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.tasky.feature_agenda.data.worker
+
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ListenableWorker
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.workDataOf
+import com.example.tasky.common.Constants
+import com.example.tasky.util.ErrorType
+import com.example.tasky.util.Resource
+
+fun enqueueWorker(
+    workManager: WorkManager,
+    inputData: Data,
+    requestBuilder: OneTimeWorkRequest.Builder
+) {
+    val constraints = Constraints.Builder()
+        .setRequiredNetworkType(NetworkType.CONNECTED)
+        .build()
+
+    val myWorkRequest = requestBuilder
+        .setConstraints(constraints)
+        .setInputData(inputData)
+        .build()
+
+    workManager.enqueue(myWorkRequest)
+}
+
+fun <T> getWorkerResult(result: Resource<T>): ListenableWorker.Result {
+    return when(result) {
+        is Resource.Error -> {
+            when(result.errorType) {
+                ErrorType.HTTP -> {
+                    ListenableWorker.Result.failure(workDataOf(Constants.WORK_DATA_KEY to result.message))
+                }
+                ErrorType.IO -> {
+                    ListenableWorker.Result.retry()
+                }
```

Comment:
If there is no internet or the server is unavailable, this will make the worker be stuck in an infinite loop. You can avoid this in the worker by adding a max retry count:
```
if(runAttemptCount > 3) {
    return Result.failure()
}
```